src/detections/by-hours.js

```diff
@@ -0,0 +1,37 @@
+import { from, reduce } from 'rxjs'
+import { mergeMap, tap } from 'rxjs/operators'
+
+import { data } from 'detections/data.json'
+
+const countDetections = value => {
+  const array = [[], [], [], []]
+  value.map(entry => array[Math.floor(new Date(entry.timestamp).getHours() / 6)].push(entry))
+
+  return array.map(entry => entry.length)
+}
+
+const addToClass = (acc, current) => {
+  acc[current.class].push(current)
+
+  return acc
+}
+
+const addClass = (acc, current) => {
+  acc[current.class] = [current]
+
+  return acc
+}
+
+const accumulator = (acc, [key, value]) => {
+  acc[key] = countDetections(value)
+
+  return acc
+}
+
+export const ByHours = setState => from(data.detections.items)
```

Comment:
Also, all these functions are not components, so let's not use capital letters here.

Reply:
done